scripts/deploymentextensions/enablemobileapi/install.sh

```diff
@@ -0,0 +1,394 @@
+
+# List the VMSSs in the target cluster. 
+# If the user specifies a VMSS filter (the deploymentId), list only that VMSS. Otherwise, list all VMSSs.
+log "Getting the list of VMSS in the '${azure_resource_group}' resource group"
+vmssIdsArray=(`az vmss list --resource-group ${azure_resource_group} | jq -r '.[] | .id'`)
+exit_on_error "Could not get a list of the VMSSs in the '${azure_resource_group}' resource group from '${HOSTNAME}' !" "${error_mobilerestapi_update_failed}" "${notification_email_subject}" "${cluster_admin_email}"
+
+vmssIdsList=""
+for vmssId in "${vmssIdsArray[@]}"
+do
+    if [[ -z $vmss_deployment_id ]] || ( [[ ! -z $vmss_deployment_id ]] && [[ $vmssId == *"vmss-$vmss_deployment_id" ]] );
+    then
+        log "Adding ${vmssId}"
+
+        # the user has specified a filter, apply it
+        vmssIdsList="${vmssId} ${vmssIdsList}"
+    fi
+done
+
+# Convert the list into an array for further processing
+filteredVmssIdsArray=($vmssIdsList)
+
+# Provide quick summary
+log "${#vmssIdsArray[@]} VMSS(s) identified for processing. After optional filtering, ${#filteredVmssIdsArray[@]} VMSS(s) will be targeted for update"
+
+# The structure of the response differ when multiple VMSSs are involved. Therefore, we have to account for that.
+log "Listing all VM instances in the targeted VMSS(s)"
+vmssInstanceList=`az vmss list-instances --resource-group ${azure_resource_group} --ids ${vmssIdsList}`
+exit_on_error "Could not get a list of the VMs in the identified VMSSs' !" "${error_mobilerestapi_update_failed}" "${notification_email_subject}" "${cluster_admin_email}"
+
+if [[ ${#filteredVmssIdsArray[@]} -gt 1 ]];
+then
+    vmssNicsArray=(`echo $vmssInstanceList | jq -r '.[][] .networkProfile.networkInterfaces[] .id'`)
+else
+    vmssNicsArray=(`echo $vmssInstanceList | jq -r '.[] .networkProfile.networkInterfaces[] .id'`)
+fi
+exit_on_error "Could not process the vmss instance list!" "${error_mobilerestapi_update_failed}" "${notification_email_subject}" "${cluster_admin_email}"
+
+# The final list of Ip addresses associated with the VMs in the targeted VMSS(s).
+vmssIpsMasterList=""
+
+# Iterate all NICs only add unique value to the array
+log "Getting all relevant VMSS nics"
+
+for vmssNic in "${vmssNicsArray[@]}"
+do
+    vmssIpsList=`az vmss nic list --ids "${vmssNic}" | jq -r '.[] .ipConfigurations[] .privateIpAddress'`
+    exit_on_error "Could not get a list of the IPs for the identified VMSSs' !" "${error_mobilerestapi_update_failed}" "${notification_email_subject}" "${cluster_admin_email}"
+
+    # add to the master list
+    vmssIpsMasterList="${vmssIpsMasterList} ${vmssIpsList}"
+done
+
+# The master list will have duplicates. Prune it.
+vmssIpsArray=(`echo $vmssIpsMasterList | tr ' ' '\n' | sort -u`)
+log "${#vmssIpsArray[@]} IP(s) discovered. Iterating each instance to enable the Mobile Rest Api."
+
+# Iterate the vmss intances and enable mobile rest api
+for vmssInstanceIp in "${vmssIpsArray[@]}"
+do
```

Comment:
nit: we generally put a semicolon ; between before "do" (and also "then"). i don't believe this is required when we put "do" or "then" on their own line, but it'd be good to be consistent.
fwiw: i prefer "then" and "do" appear on the same line as the if condition or loop statement